src/base/types.h

```diff
@@ -113,12 +113,20 @@ typedef unordered_map<JobID_t, JobDescriptor,
         boost::hash<boost::uuids::uuid> > JobMap_t;
 typedef unordered_map<JobID_t, TaskGraph*,
         boost::hash<boost::uuids::uuid> > TaskGraphMap_t;*/
+typedef thread_safe::set<ResourceID_t> ResourceSet_t;
+typedef thread_safe::map<ResourceID_t, uint64_t> ResourceFrequencyMap_t;
+typedef thread_safe::map<ResourceID_t, EquivClass_t> ResourceEquivClassMap_t;
+typedef thread_safe::map<ResourceID_t, int64_t> ResourceCostMap_t;
 typedef thread_safe::map<ResourceID_t, ResourceStatus*> ResourceMap_t;
 typedef thread_safe::map<JobID_t, JobDescriptor> JobMap_t;
 typedef thread_safe::map<JobID_t, TaskGraph*> TaskGraphMap_t;
 #else
 typedef uint64_t ResourceID_t;
 typedef uint64_t JobID_t;
+typedef unordered_set<ResourceID_t> ResourceSet_t;
+typedef unordered_map<ResourceID_t, uint64_t> ResourceFrequencyMap_t;
```

Comment:
Only used in one place (simulated_quincy_cost_model). I would not define it in types if it's only used in one place.